ethrpc2/ethrpc.go

```diff
@@ -0,0 +1,329 @@
+package ethrpc2
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/big"
+
+	"github.com/0xsequence/ethkit/ethrpc2/jsonrpc"
+	"github.com/0xsequence/ethkit/go-ethereum"
+	"github.com/0xsequence/ethkit/go-ethereum/common"
+	"github.com/0xsequence/ethkit/go-ethereum/common/hexutil"
+	"github.com/0xsequence/ethkit/go-ethereum/core/types"
+	"github.com/0xsequence/ethkit/go-ethereum/rpc"
+)
+
+func ChainID() CallBuilder[*big.Int] {
```

Comment:
any reason why these methods should be exported as they feel mostly needed for internal use within the lib?

Reply:
nice, makes sense :)